src/sessions.ts

```diff
@@ -286,12 +288,12 @@ export class ClientSession extends TypedEventEmitter<ClientSessionEvents> {
    */
   startTransaction(options?: TransactionOptions): void {
     if (this[kSnapshotEnabled]) {
-      throw new MongoDriverError('Transactions are not allowed with snapshot sessions');
+      throw new MongoTransactionError('Transactions are not allowed with snapshot sessions');
     }
 
     assertAlive(this);
     if (this.inTransaction()) {
-      throw new MongoDriverError('Transaction already in progress');
+      throw new MongoTransactionError('Transaction already in progress');
```

Comment:
This is also kind of an api error; we may need to rethink how we draw the line between runtime and api errors, specifically with respect to transactions: like, it's true that the driver is getting to an invalid state, but it's not "unexpected" because it's still the user making a choice; maybe the correct thing to do is subclass the transaction errors from API errors, or to subclass runtime errors into incorrect usage ones vs unexpected driver state, and then subclass these transaction errors according to the cause of the error

Reply:
Capturing the rebucketing that happened in NODE-3421 where errors have been switched between MongoRuntimeError and MongoAPIError, with MongoTransactionError being one of them. MongoTransactionError is now a child of MongoAPIError.